### ISSUE TITLE
How to use DeepVAR Model with a multivariate time sereis

### ISSUE THREAD

**Issue by FunkyFabe** [jdb78/pytorch-forecasting#1295]
- PyTorch-Forecasting version: 1.0.0
- PyTorch version:  Latest
- Python version: 3.10
- Operating System: macOS Ventura

I have experience in time series forecasting using the Darts library.  Now, I want to train a [DeepAR](https://pytorch-forecasting.readthedocs.io/en/stable/api/pytorch_forecasting.models.deepar.DeepAR.html) model in combination with the 
[MultivariateNormalDistributionLoss](https://pytorch-forecasting.readthedocs.io/en/stable/api/pytorch_forecasting.metrics.distributions.MultivariateNormalDistributionLoss.html) using PyTorch Forecasting. I have a multivariate time series with 4 dimensions (see code below). By multivariate I mean that the time series is vector-valued. However, the concepts in PyTorch Forecasting are not quite clear to me and I encountered some strange beheaviour. 

How can I create a multivariate time series in Pytorch Forecasting? Do I have to use `group_ids` or do I have to use multiple columns (multiple `targets`). To get started, I followed the [DeepVAR tutorial](https://pytorch-forecasting.readthedocs.io/en/stable/tutorials/deepar.html). This seems to me the correct way to define a multivaraite time series. However, when I insepcted the  `x['target_scale']` of a sample, I saw that all dimensions are scaled with the same mean and standard deviation, which are coming from all values of all dimensions. Shouldn't there be a scaler per dimension? To better understand the problem, I put my code to create the TimeSeriesDataSets and DataLoaders below.

**x['target_scale']**
```
tensor([[1.9450, 1.1488],
        [1.9450, 1.1488],
        [1.9450, 1.1488],
        [1.9450, 1.1488]])
```

```
N_SERIES = 4
N_STEPS = 100
base_values = np.arange (0, 1, 1/N_STEPS)
values = [v+s for s in range(0,N_SERIES) for v in  base_values]
series = [s for s in range(0, N_SERIES) for _ in range(0,100)]
time_idx = [_ for s in range(0, N_SERIES) for _ in range(0,100)]
data = pd.DataFrame({
    'series': series,
    'time_idx': time_idx,
    'value': values
})
data = data.astype(dict(series=str))

# create dataset and dataloaders
max_encoder_length = 20
max_prediction_length = 10

training_cutoff = data["time_idx"].max() - max_prediction_length

context_length = max_encoder_length
prediction_length = max_prediction_length

training = TimeSeriesDataSet(
    data[lambda x: x.time_idx <= training_cutoff],
    time_idx="time_idx",
    target="value",
    categorical_encoders={"series": NaNLabelEncoder().fit(data.series)},
    group_ids=["series"],
    static_categoricals=[
        "series"
    ],  # as we plan to forecast correlations, it is important to use series characteristics (e.g. a series identifier)
    time_varying_unknown_reals=["value"],
    max_encoder_length=context_length,
    max_prediction_length=prediction_length,
)

validation = TimeSeriesDataSet.from_dataset(training, data, min_prediction_idx=training_cutoff + 1)

batch_size = 4
# synchronize samples in each batch over time - only necessary for DeepVAR, not for DeepAR
train_dataloader = training.to_dataloader(
    train=True, batch_size=batch_size, num_workers=0, batch_sampler="synchronized"
)

val_dataloader = validation.to_dataloader(
    train=False, batch_size=batch_size, num_workers=0, batch_sampler="synchronized"
)
```

**Comment by khairulislam** [comment]
`group_ids` is for multiple time series data. For example, in your case, each value in the `range(0, N_SERIES)` denotes one time series. If multivariate, we would pass multiple input/output variables to the model. From your code, you only have one target feature `values`. If you have multiple static, dynamic, or target features then you can pass those multiple columns in the `TimeSeriesDataSet`. 
Also, I believe this library appends the past target values in the encoder too. So `time_varying_unknown_reals=["value"]` would not be necessary.

**Comment by FunkyFabe** [comment]
Hi @khairulislam ,
Thank you for explaining the mechanism of `group_ids` and multiple target features and sharing the TFT example with me. If I define multiple target featues, can I still use the  MultivariateNormalDistributionLoss with the DeepAR model to account correlation between the dimensions?

**Comment by khairulislam** [comment]
You are welcome. The MND loss will still be usable. For multiple targets, you might need to write something like the bellow 
```python
from pytorch_forecasting.metrics import MultiLoss

net = DeepAR.from_dataset(
    training,
    learning_rate=3e-2,
    hidden_size=30,
    rnn_layers=2,
    loss=MultiLoss([MultivariateNormalDistributionLoss(rank=30) for _ in targets]), 
    optimizer="Adam",
)
```
The `MultiLoss`  is a metric that can be used to combine multiple metrics.
